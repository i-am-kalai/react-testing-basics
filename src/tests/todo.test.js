import { render, screen, cleanup } from "@testing-library/react";
import Todo from "../components/todo";
import renderer from "react-test-renderer";

afterEach(() => { cleanup() })

test('should render non-completed todo', () => {
    const testTodo = {
        id: 'todo-1',
        title: 'Learn testing',
        isCompleted: false
    };
    render(<Todo todo={testTodo} />);
    const todoElement = screen.getByTestId(testTodo.id)
    expect(todoElement).toBeInTheDocument()
    expect(todoElement).toHaveTextContent(testTodo.title)
    expect(todoElement).not.toContainHTML('<strike>')
})

test('should render completed todo', () => {
    const testTodo = {
        id: 'todo-2',
        title: 'Do well',
        isCompleted: true
    };
    // Just another way of querying rendered elements instead of using 'screen'
    const todoElement = render(<Todo todo={testTodo} />).queryByTestId(testTodo.id);
    expect(todoElement).toBeInTheDocument()
    expect(todoElement).toHaveTextContent(testTodo.title)
    expect(todoElement).toContainHTML('strike')
})

test('match snapshot', () => {
    const testTodo = {
        id: 'todo-2',
        title: 'Do well',
        isCompleted: true
    };
    const tree = renderer.create(<Todo todo={testTodo} />).toJSON()
    expect(tree).toMatchSnapshot()
})