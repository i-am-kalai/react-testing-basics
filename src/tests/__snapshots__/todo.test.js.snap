// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  data-testid="todo-2"
>
  <strike>
    <h1>
      Do well
    </h1>
  </strike>
</div>
`;
